### PR TITLE
Add asset listing fee

### PR DESCRIPTION
**Pull request by ManfredKarrer**

### CODE REVIEW

**Comment by ManfredKarrer** [comment]
See https://github.com/bisq-network/proposals/issues/42 for context...

I uploaded here a video to demonstrate the implementation:
https://youtu.be/mUQOKs4Va1w

File: core/src/main/java/bisq/core/dao/state/model/blockchain/OpReturnType.java [hunk 1]
```diff
@@ -37,7 +37,9 @@
     REIMBURSEMENT_REQUEST((byte) 0x12),
     BLIND_VOTE((byte) 0x13),
     VOTE_REVEAL((byte) 0x14),
-    LOCKUP((byte) 0x15);
+    LOCKUP((byte) 0x15),
+    ASSET_LISTING_FEE((byte) 0x16),
+    PROOF_OF_BURN((byte) 0x17);
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/state/model/blockchain/OpReturnType.java:42]:
```suggestion
    LOCKUP((byte) 0x17);
```

File: core/src/main/java/bisq/core/dao/state/model/blockchain/OpReturnType.java [hunk 2]
```diff
@@ -37,7 +37,9 @@
     REIMBURSEMENT_REQUEST((byte) 0x12),
     BLIND_VOTE((byte) 0x13),
     VOTE_REVEAL((byte) 0x14),
-    LOCKUP((byte) 0x15);
+    LOCKUP((byte) 0x15),
+    ASSET_LISTING_FEE((byte) 0x16),
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/state/model/blockchain/OpReturnType.java:41]:
```suggestion
    PROOF_OF_BURN((byte) 0x16),
```

File: core/src/main/java/bisq/core/dao/state/model/blockchain/OpReturnType.java [hunk 3]
```diff
@@ -37,7 +37,9 @@
     REIMBURSEMENT_REQUEST((byte) 0x12),
     BLIND_VOTE((byte) 0x13),
     VOTE_REVEAL((byte) 0x14),
-    LOCKUP((byte) 0x15);
+    LOCKUP((byte) 0x15),
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/state/model/blockchain/OpReturnType.java:40]:
```suggestion
    ASSET_LISTING_FEE((byte) 0x15),
```

File: core/src/main/java/bisq/core/dao/governance/asset/AssetService.java [hunk 4]
```diff
@@ -17,99 +17,336 @@
...
-    public boolean hasPaidBSQFee(String tickerSymbol) {
-        //TODO
-        return false;
+    @Override
+    public void onParseTxsComplete(Block block) {
+        int chainHeight = daoStateService.getChainHeight();
+        bsqFeePerDay = daoStateService.getParamValueAsCoin(Param.ASSET_LISTING_FEE_PER_DAY, chainHeight).value;
+        minVolumeInBtc = daoStateService.getParamValueAsCoin(Param.ASSET_MIN_VOLUME, chainHeight).value;
+        updateList();
+
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/governance/asset/AssetService.java:272]:
```suggestion

```

**Review by sqrrm — APPROVED** [review]
utAck

**Comment by ManfredKarrer** [comment]
@sqrrm The suggested changes are not clear to me from the GH view. Could you describe or post the whole suggested change code?

**Comment by sqrrm** [comment]
Hmm, yeah, thought it would be smooth to do it inline, better you do it. I just noticed the PROOF_OF_BURN and ASSET_LISTING_FEE coming before lockup in most enums so I tried to make it consistent, not very clear perhaps in my suggested changes.

**Comment by ManfredKarrer** [comment]
Its only in TxOutputType but there its opReturn types which are all grouped. All others are as well after lockup.

**Pull request merged**